engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/VectorBenchmarks.java

```diff
@@ -0,0 +1,142 @@
+package org.enso.interpreter.bench.benchmarks.semantic;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.Value;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.BenchmarkParams;
+import org.openjdk.jmh.infra.Blackhole;
+
+
+@BenchmarkMode(Mode.AverageTime)
+@Fork(1)
+@Warmup(iterations = 3)
+@Measurement(iterations = 5)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+public class VectorBenchmarks {
+  private Value arrayOfFibNumbers;
+  private Value avg;
+  private Value self;
+
+  @Setup
+  public void initializeBenchmark(BenchmarkParams params) {
+    Engine eng = Engine.newBuilder()
+      .allowExperimentalOptions(true)
+      .logHandler(new ByteArrayOutputStream())
+      .option(
+        "enso.languageHomeOverride",
+        Paths.get("../../distribution/component").toFile().getAbsolutePath()
+      ).build();
+    var ctx = Context.newBuilder()
+      .engine(eng)
+      .allowIO(true)
+      .allowAllAccess(true)
+      .build();
+    var module = ctx.eval("enso", "\n" +
+      "import Standard.Base.Data.Vector\n" +
+      "\n" +
+      "avg arr =\n" +
+      "    sum acc i = if i == arr.length then acc else\n" +
+      "        sum (acc + arr.at i) i+1\n" +
+      "    (sum 0 0) / arr.length\n" +
+      "\n" +
+      "fibarr size modulo = \n" +
+      "    b = Vector.new_builder size\n" +
```

Comment:
Not in bechmarks. CCing @hubertp

Reply:
Yes, this is known. The way benchmark task is setup is a) unnecessarily complex making it hard for Frgaal integration b) there were some non-trivial problems once I managed to setup it up but I can't recall exactly what now.
So at the time we decided that it wasn't worth spending time on it since people won't ask for it (famous last words).